presto-druid/src/main/java/com/facebook/presto/druid/DruidSplitManager.java

```diff
@@ -52,16 +69,119 @@ public ConnectorSplitSource getSplits(
     {
         DruidTableLayoutHandle layoutHandle = (DruidTableLayoutHandle) layout;
         DruidTableHandle table = layoutHandle.getTable();
-        if (isComputePushdownEnabled(session) || (table.getDql().isPresent() && table.getDql().get().getPushdown())) {
+        if (isComputePushdownEnabled(session) && (table.getDql().isPresent() && table.getDql().get().getPushdown())) {
```

Comment:
In order for the code to use the segment approach, I had to put the && such the the session property when set to false is allows it to do a segment approach vs. broker approach.